test/main.jl

```diff
@@ -36,4 +38,55 @@ withenv("DATADEPS_ALWAYS_ACCEPT"=>"true") do
         @test true
     end
 
+
+    @testset "Ensure when errors occur the datadep will still retrydownloading" begin
+        @testset "error in checksum" begin
+            @stub dummydown
+            @expect dummydown(::Any, ::Any) = joinpath(@__DIR__, "eg.zip")
+            register(DataDep("TestErrorChecksum", "dummy message", "http://example.void",
+                             (error, "1234"); # this will throw an error
+                             fetch_method=dummydown))
+            @test_throws Exception datadep"TestErrorChecksum"
```

Comment:
not super important, but you could use an `ErrorException` here to be just a tiny bit more confident that you are catching the error you expect to catch

Reply:
Good point. I remember thinking that. I don't know why I didn't